src/state/render_pipeline.rs

```diff
@@ -1,118 +1,61 @@
-use super::buffer;
-use super::vertex::Vertex;
+use super::{vertex::Vertex, State};
 
-pub struct RenderPipeline {
-    render_pipeline: wgpu::RenderPipeline,
-}
-
-impl RenderPipeline {
-    pub fn new(
+impl State {
```

Comment:
Same question: why do we need to imp State when it is just free function 🤔 ?

Reply:
Refactored. 
`render_pipeline.rs` is now gone. All `config` functions go to `state/builder/init.rs`